Libraries/LibJS/Parser.cpp

```diff
@@ -314,11 +332,13 @@ NonnullRefPtr<Statement> Parser::parse_statement()
                 return result.release_nonnull();
         }
         if (match_expression()) {
+            if (match(TokenType::Function))
+                syntax_error("Function declaration not allowed in single-statement context");
             auto expr = parse_expression(0);
             consume_or_insert_semicolon();
             return create_ast_node<ExpressionStatement>(move(expr));
         }
-        expected("statement (missing switch case)");
+        expected("statement");
```

Comment:
I decided to get rid of these as this is not really the case anymore - it's not a "missing" switch case like in the early days, it's a syntax error, period.